emails/models.py

```diff
@@ -195,19 +194,26 @@ def fxa_locale_in_premium_country(self):
     def avatar(self):
         return self.fxa.extra_data.get("avatar")
 
-    @cached_property
+    @property
     def relay_addresses(self):
-        # TODO: Remove cached_property, to avoid cache invalidation
         return RelayAddress.objects.filter(user=self.user)
 
-    @cached_property
+    @property
     def domain_addresses(self):
-        # TODO: Remove cached_property, to avoid cache invalidation
         return DomainAddress.objects.filter(user=self.user)
 
     @property
-    def num_active_address(self):
-        return len(self.relay_addresses) + len(self.domain_addresses)
+    def total_masks(self) -> int:
+        ra_count: int = self.relay_addresses.count()
+        da_count: int = self.domain_addresses.count()
+        return ra_count + da_count
+
+    @property
+    def at_mask_limit(self) -> bool:
+        if self.has_premium:
+            return False
+        ra_count: int = self.relay_addresses.count()
+        return ra_count == settings.MAX_NUM_FREE_ALIASES
```

Comment:
Let's make this work for downgraded users too - they had premium, and now they are on the free plan, but have over the limit:

```suggestion
        return ra_count >= settings.MAX_NUM_FREE_ALIASES
```